platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/roster/RosterRetriever.java

```diff
@@ -158,7 +162,21 @@ public static RosterEntry buildRosterEntry(
             final long weight,
             @NonNull final Bytes cert,
             @NonNull final List<ServiceEndpoint> endpoints) {
-        final List<ServiceEndpoint> rosterOrdered = endpoints.size() > 1 ? endpoints.reversed() : endpoints;
+        // Filter out any invalid endpoints
+        final List<ServiceEndpoint> filtered = endpoints.stream()
+                .filter(e -> {
```

Comment:
Probably. What is the expected behavior in that case?

Reply:
Nope, there is no code to handle that.    Can't connect to anyone that has 0 gossip endpoints.  Its fundamentally bad input to be able to reach this state.

Reply:
@anthony-swirldslabs  wrote the original.     If we have proper input validation, this filter should never execute.

Reply:
Both `NodeCreateHandler` and `NodeUpdateHandler` call `AddressBookValidator#validateServiceEndpoint` to verify a valid endpoint ([here](https://github.com/hashgraph/hedera-services/blob/778d2794ca3f60be28a864538484659d348d3afd/hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeCreateHandler.java#L109) and [here](https://github.com/hashgraph/hedera-services/blob/778d2794ca3f60be28a864538484659d348d3afd/hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeUpdateHandler.java#L130)), so we should be good 👍

Reply:
Likely, "we just missed it back then". Less likely, but not impossible, some unit tests might use bogus AddressBooks w/o any endpoints at all, in which case this code would fail to support them. If the latter is true, well, then we're good. If not though, I think it wouldn't hurt adding the exception while we're refactoring this code anyway.